src/java/org/apache/cassandra/db/filter/RowFilter.java

```diff
@@ -1081,6 +1106,112 @@ protected Kind kind()
         }
     }
 
+
+    // VSTODO make generic?
+    public static class GeoDistanceExpression extends Expression
+    {
+        private final ByteBuffer distance;
+        private final Operator distanceOperator;
+        private final float searchRadiusMeters;
+        private final float searchRadiusDegreesSquared;
+        private final float searchLat;
+        private final float searchLong;
+
+        public GeoDistanceExpression(ColumnMetadata column, ByteBuffer point, Operator operator, ByteBuffer distance)
+        {
+            super(column, Operator.BOUNDED_ANN, point);
+            assert column.type instanceof VectorType && (operator == Operator.LTE || operator == Operator.LT);
+            this.distanceOperator = operator;
+            this.distance = distance;
+            searchRadiusMeters = FloatType.instance.compose(distance);
+            if (searchRadiusMeters <= 0)
+                throw new InvalidRequestException("GEO_DISTANCE radius must be non-negative, got " + searchRadiusMeters);
```

Comment:
do we want to validate a max radius too, possibly as a guardrail?

Reply:
I will follow up with this in another PR.

Reply:
What is a good default limit @jbellis?

Reply:
You know as much as I do. :)